src/ws/service/friendships_service.rs

```diff
@@ -265,3 +300,196 @@ pub fn map_request_events(requests: Vec<FriendshipRequestEvent>, user_id: String
         }),
     }
 }
+
+async fn process_room_event_ws(
+    event: FriendshipEventPayload,
+    db: DatabaseComponent,
+    user_id: String,
+) {
+    // Get event
+
+    // Get user from event
+    let address_1 = user_id;
+    let address_2 = "".to_string(); // Get it from the event
+
+    // Get the friendship info
+    let db_repos = &db.clone().db_repos.unwrap();
+    let friendships_repository = &db_repos.friendships;
+    let friendship = get_friendship(friendships_repository, &address_1, &address_2)
+        .await
+        .unwrap();
+
+    //  Get the last status from the database to later validate if the current action is valid.
+    let friendship_history_repository = &db_repos.friendship_history;
+    let _last_history = get_last_history(&friendship, friendship_history_repository)
+        .await
+        .unwrap();
+
+    // Validate if the new status that is trying to be set is valid. If it's invalid or it has not changed, return here.
```

Comment:
Nope, I don't think it would be a valid event. It cannot even be done from the UI, so if it does occur, it's more likely related to a bug than a wanted action ✨